pkg/csi/cinder/controllerserver.go

```diff
@@ -58,14 +60,32 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		return nil, err
 	}
 
-	// Volume Create
-	resID, resAvailability, err := cloud.CreateVolume(volName, volSizeGB, volType, volAvailability, nil)
+	// Verify a volume with the provided name doesn't already exist for this tenant
+	volumes, err := cloud.GetVolumesByName(volName)
```

Comment:
It is better to check if  ```volSizeGB```,```volType```,```volAvailability``` of the ```volumes``` are equal with the same fields in the```CreateVolumeRequest```.

Reply:
Based on @j-griffith's description of the spec, I think it's helpful for debugging but not compliant with the requirements. If identity is equivalent to name, then name should be the only criteria we use for identity.